packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -937,7 +942,23 @@ contract VotingV2 is
             voterStake.lastRequestIndexConsidered = requestIndex + 1;
         }
 
-        if (slash != 0) _applySlashToVoter(slash, voterStake, voterAddress);
+        // If there is any remaining slashing then apply it. This occurs when there is unapplied slashing in the loop
+        // due to the last unlashed elements all being all from the same round. i.e we only slash within the loop when
+        // transitioning between rounds and the last round is slashed here. Note that there is a special case that needs
+        // to be considered separately: if the indexTo is less than the priceRequestIndex.length then we know we
+        // are dealing with a partial index slashing via a call to updateTrackersRange. If this is the case, we need to
+        // be careful for ranges that bisect a voting round as all slashing must be applied linearly and independently
+        // within each round. If we are in this case and the element before the indexTo (last element executed in the
+        // loop) has the same voting round as the indexTo (element after the last element in the loop) then we know that
+        // we've bisected a round and should store the unapplied slashing which will seed this method on the next entry
+        // such that the slashing will be applied linearly, not compounding with other slashing within the same round.
+        if (slash != 0)
+            if (
+                indexTo < priceRequestIds.length &&
+                priceRequests[priceRequestIds[indexTo - 1]].lastVotingRound ==
+                priceRequests[priceRequestIds[indexTo]].lastVotingRound
```

Comment:
I've been looking at this logic for a while, and I can't see how this avoids the problem where indexTo - 1 is a rolled vote. Couldn't that show a change in roundId when there wasn't supposed to be one?

I could be missing something obvious, but I just don't see how all this logic works in the case where indexTo < priceRequestIds.length.